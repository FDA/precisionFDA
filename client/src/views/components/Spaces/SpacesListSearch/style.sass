.spaces-list-search

  &__input
    width: 500px
    position: relative

  &__icon
    cursor: pointer
    position: absolute
    top: 7px
    right: 10px
