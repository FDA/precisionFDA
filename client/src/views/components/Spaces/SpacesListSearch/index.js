import React, { useCallback, useState } from 'react'
import PropTypes from 'prop-types'

import { debounce } from '../../../../utils'
import Icon from '../../Icon'
import './style.sass'


const SpacesListSearch = ({ filterSpacesHandler }) => {
  const [spacesListSearch, setSpacesListSearch] = useState('')
  const showIcon = !!spacesListSearch.length

  const debFilter = useCallback(debounce((value) => filterSpacesHandler(value), 400), [])

  const onChange = (e) => {
    setSpacesListSearch(e.target.value)
    debFilter(e.target.value)
  }

  const clearSearch = () => {
    setSpacesListSearch('')
    filterSpacesHandler('')
  }

  return (
    <div className="spaces-list-search">
      <div className="spaces-list-search__input">
        <input
          name="spacesListSearch"
          value={spacesListSearch}
          placeholder="Filter by State, Name, Type or User..."
          onChange={onChange}
          type="text"
          autoComplete="off"
          className="form-control"
          aria-label="Search box to filter Spaces by State, Name, Type or User..."
        />
        {(showIcon) && (
          <div className="spaces-list-search__icon">
            <Icon icon="fa-times" onClick={clearSearch} />
          </div>
        )}
      </div>
    </div>
  )
}

SpacesListSearch.propTypes = {
  filterSpacesHandler: PropTypes.func,
}

export default SpacesListSearch
