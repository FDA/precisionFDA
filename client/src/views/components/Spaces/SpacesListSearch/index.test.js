import React from 'react'
import { shallow } from 'enzyme'

import SpacesListSearch from './'


describe('<SpacesListSearch />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<SpacesListSearch />)
    expect(wrapper).toMatchSnapshot()
  })
})
