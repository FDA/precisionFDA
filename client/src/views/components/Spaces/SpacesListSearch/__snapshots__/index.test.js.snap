// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpacesListSearch /> matches snapshot 1`] = `
<div
  className="spaces-list-search"
>
  <div
    className="spaces-list-search__input"
  >
    <input
      autoComplete="off"
      className="form-control"
      name="spacesListSearch"
      onChange={[Function]}
      placeholder="Filter by State, Name, Type or User..."
      type="text"
      value=""
    />
  </div>
</div>
`;
